apps/opik-backend/src/main/java/com/comet/opik/api/DatasetItem.java

```diff
@@ -65,4 +89,19 @@ public static class Write {
         public static class Public {
         }
     }
+
+    public DatasetItem {
+        if (data != null
+                && data.values().stream().filter(Objects::nonNull).anyMatch(not(value -> value instanceof JsonNode))) {
+            throw new IllegalArgumentException("Data field must be an Map<String, JsonNode>");
```

Comment:
Do you think this should be a 400? The Deserializer should guarantee that the Map will always contain JsonNode values. The only way this would happen is when creating the object manually. That is way I throw `IllegalArgumentException`